core/packages/contracts/contracts/Vault.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// TODO: transfer ownership from deployer to SovereignTreasury
+// This contract actually holds Ether balances for each sovereignID.
+contract Vault is Ownable {
```

Comment:
Another option here is a [PullPayment](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/PullPayment.sol). This would also prevent malicious recipients from blocking execution: a potential attack is using up all remaining gas on a withdrawal, blocking all future messages for a given sourceID/parachain.

This isn't much of a concern though, given that sources/parachains will control the `recipient` of Withdraw actions and the impact is isolated to their ability to use the bridge.

It might be a consideration for dev experience if the recipient account can be a contract controlled by a source/parachain, but for now we're assuming it's an EOA.